### PR TITLE
Address warnings and deprecations

### PR DESCRIPTION
**Pull request by msbarry**
- refer directly to nested classes in ForwardingProfile (instead of through OpenMapTilesProfile)
- use vector tile feature `tags()` method instead of deprecated `attrs()`
- reduce verbosity of bad water polygon logging
- remove some unused variables

### CODE REVIEW

File: src/main/java/org/openmaptiles/layers/Water.java [hunk 1]
```diff
@@ -201,7 +201,7 @@ void attemptNeLakeIdMapping(Tables.OsmWaterPolygon element) throws GeometryExcep
     if (!geom.isValid()) {
       geom = GeometryFixer.fix(geom);
       stats.dataError("omt_fix_water_before_ne_intersect");
-      LOGGER.debug("Fixing geometry of OSM element {} before attempt to add ID to natural earth water feature",
+      LOGGER.trace("Fixing geometry of OSM element {} before attempt to add ID to natural earth water feature",
```
Comment (msbarry) [inline src/main/java/org/openmaptiles/layers/Water.java:204]: @phanecak-maptiler these are pretty verbose - any issue with me turning them to trace so they are not printed by default?

File: src/main/java/org/openmaptiles/Generate.java [hunk 2]
```diff
@@ -604,16 +604,16 @@ private static List<OsmTableField> parseTableFields(Imposm3Table tableDefinition
    * "class") based on the "field mapping" defined in the layer schema definition.
    */
   static MultiExpression<String> generateFieldMapping(JsonNode valuesNode) {
-    MultiExpression<String> mapping = MultiExpression.of(new ArrayList<>());
+    List<MultiExpression.Entry<String>> mappings = new ArrayList<>();
```
Comment (msbarry) [inline src/main/java/org/openmaptiles/Generate.java:607]: I want to change MultiExpression to make an immutable copy of the input, this code was assuming it is mutable.

File: src/main/java/org/openmaptiles/layers/Water.java (hunk 1 above)
Comment (msbarry) [reply src/main/java/org/openmaptiles/layers/Water.java:204]: I'll merge like this for now, feel free to move one or both back up to debug if you decide you need it.

Comment (phanecak-maptiler) [reply src/main/java/org/openmaptiles/layers/Water.java:204]: No problem. Thank you.

File: src/main/java/org/openmaptiles/Generate.java (hunk 2 above)
Comment (phanecak-maptiler) [reply src/main/java/org/openmaptiles/Generate.java:607]: OK.

**Pull request merged**